src/Adapter/Product/Combination/CommandHandler/UpdateCombinationFromListingHandler.php

```diff
@@ -0,0 +1,113 @@
+use PrestaShop\PrestaShop\Adapter\Product\Combination\Update\DefaultCombinationUpdater;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Command\UpdateCombinationFromListingCommand;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\CommandHandler\UpdateCombinationFromListingHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Product\Combination\Exception\CannotUpdateCombinationException;
+
+/**
+ * Handles @see UpdateCombinationFromListingCommand using legacy object model
+ */
+final class UpdateCombinationFromListingHandler implements UpdateCombinationFromListingHandlerInterface
+{
+    /**
+     * @var CombinationRepository
+     */
+    private $combinationRepository;
+
+    /**
+     * @var DefaultCombinationUpdater
+     */
+    private $defaultCombinationUpdater;
+
+    /**
+     * @var CombinationStockUpdater
+     */
+    private $combinationStockUpdater;
+
+    /**
+     * @param CombinationRepository $combinationRepository
+     * @param DefaultCombinationUpdater $defaultCombinationUpdater
+     * @param CombinationStockUpdater $combinationStockUpdater
+     */
+    public function __construct(
+        CombinationRepository $combinationRepository,
+        DefaultCombinationUpdater $defaultCombinationUpdater,
+        CombinationStockUpdater $combinationStockUpdater
+    ) {
+        $this->combinationRepository = $combinationRepository;
+        $this->defaultCombinationUpdater = $defaultCombinationUpdater;
+        $this->combinationStockUpdater = $combinationStockUpdater;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(UpdateCombinationFromListingCommand $command): void
+    {
+        $combination = $this->combinationRepository->get($command->getCombinationId());
+        $updatableProperties = $this->fillUpdatableProperties($combination, $command);
+        $this->combinationRepository->partialUpdate($combination, $updatableProperties, CannotUpdateCombinationException::FAILED_UPDATE_LISTED_COMBINATION);
+
+        $this->combinationStockUpdater->update($combination, $updatableProperties);
+
+        if (true === $command->isDefault()) {
+            $this->defaultCombinationUpdater->setDefaultCombination($command->getCombinationId());
+        }
+    }
+
+    /**
+     * @param Combination $combination
+     * @param UpdateCombinationFromListingCommand $command
+     *
+     * @return array<int, string>
+     */
+    private function fillUpdatableProperties(Combination $combination, UpdateCombinationFromListingCommand $command): array
+    {
+        $updatableProperties = [];
+
+        if (null !== $command->getImpactOnPrice()) {
+            $combination->price = (float) (string) $command->getImpactOnPrice();
+            $updatableProperties[] = 'price';
+        }
+
+        if (null !== $command->getQuantity()) {
+            //@todo: should we deprecate combination->quantity and product->quantity (as we did with location?) because stock_advanced is the real source
```

Comment:
@kpodemski do you have some history about it? do you know if it's safe to say `stock_advanced` is the current standard field for quantity, and combination->quantity and product->quantity are fields used in the past and now deprectaed (but kept for backward compatibility)?

Reply:
@matks 

yes, that's true, some ERPs are still integrated into it to put here real product quantity but from what I know this field is not really used in any part of the system anymore, replaced via `StockAvailable`